core/spike/Makefile

```diff
@@ -0,0 +1,47 @@
+# Makefile for compiling LF programs for the risc-v spike platform
```

Comment:
Is the user still supposed to copy the `Makefile` manually? Why not have the code generator do it?
The instructions here seem outdated (what is `demo.riscv`?).

Moreover, it seems like this Makefile could be turned into a CMake-compatible file, which would allow it to be integrated into our existing build system. Conceivably, the CMake-compatible file can automatically be included in the generated CMakeLists.txt by the code generator for the `spike` platform. However, I don't think this is critical at the moment for merging this.

Reply:
> Is the user still supposed to copy the Makefile manually? Why not have the code generator do it?

Since the spike files are pulled in non-conditionally, it felt strange to have a spike-specific Makefile sitting in the project root when compiling for non-spike platforms.
> what is `demo.riscv`?

default name for the executable; I wasn't sure how to specify a dynamic name so just called it `demo.riscv`.
> Moreover, it seems like this Makefile could be turned into a CMake-compatible file

This would be ideal, but I wasn't sure how to get CMake to invoke riscv64-unknown-elf-gcc. Closest I found was [this stackoverflow answer](https://stackoverflow.com/questions/45933732/how-to-specify-a-compiler-in-cmake), but it says that setting CMAKE_C_COMPILER should be avoided. How would you suggest turning this into a CMake-compatible file?

Reply:
> but I wasn't sure how to get CMake to invoke riscv64-unknown-elf-gcc.

If you set the `targetConfig.compiler =  riscv64-unknown-elf-gcc`, then the `CC` environment variable will automatically [be set](https://github.com/lf-lang/lingua-franca/blob/3a3beb070ce5a7a0ff7c1465f770352c34945c64/org.lflang/src/org/lflang/generator/c/CCmakeCompiler.java#L123). According to the StackOverflow answer, this should be enough. 

However, I'm not sure about the extra options that are needed though. Those might require adding additional things to the generated CMakeLists.txt. The best way to do it is probably by using [cmake-include](https://www.lf-lang.org/docs/handbook/target-declaration#cmake-include) to include a (hypothetical) `core/spike/CMakeExtension.txt` (e.g., by adding it to `targetConfig.cmakeIncludes` or `cmakeIncludesWithoutPath`).

Reply:
If we stick with the Makefile, would you suggest conditionally pulling it into the project root based on the "platform" target parameter, always pulling it into project root, or keeping it as-is?

Reply:
I don't have a strong preference for this, as long as the instructions are easy enough to find and follow. For example, the code generator can point to a README file or a website link after code generation is finished.